lib/services/local/generic/generic_test.go

```diff
@@ -436,6 +464,81 @@ func TestGenericListResourcesWithFilter(t *testing.T) {
 	require.Equal(t, "", nextKey)
 }
 
+func TestGenericListResourcesWithFilterForScale(t *testing.T) {
+	ctx := context.Background()
+
+	memBackend, err := memory.New(memory.Config{
+		Context: ctx,
+		Clock:   clockwork.NewFakeClock(),
+	})
+	require.NoError(t, err)
+
+	service, err := NewService(&ServiceConfig[*testResource]{
+		Backend:       memBackend,
+		ResourceKind:  "generic resource",
+		PageLimit:     200,
+		BackendPrefix: backend.NewKey("my-prefix"),
+		UnmarshalFunc: unmarshalResource,
+		MarshalFunc:   marshalResource,
+	})
+	require.NoError(t, err)
+
+	totalResourcesPerProp := 100
+	totalProps := 100
+	var totalResources []*testResource
+	for i := 0; i < totalResourcesPerProp; i++ {
+		for j := 0; j < totalProps; j++ {
+			r := newTestResourceWithSpec(uuid.NewString(), strconv.Itoa(j))
+			totalResources = append(totalResources, r)
+		}
+	}
+
+	r := rand.New(rand.NewPCG(uint64(82), uint64(123)))
```

Comment:
Do we need the determinism here?

Reply:
Generally, I think having a different value every time would be better to exercise other possible combinations.
I'm confident about this one because, locally, I was testing much more combinations. I just couldn't commit them because it was slow and would trip the flaky test detector.

As for why, we have determinism here: https://github.com/gravitational/teleport/pull/47326#discussion_r1792067076